// dingtalk-design-miniapp/lib/demo/index/page.js
"use strict";

// Dingtalk design miniapp 配置

module.exports = [{
  title: '通用',
  type: 'general',
  cardList: [{
    name: '按钮',
    enname: 'button',
    id: 'button',
    link: '/src/core/button/demo/index'
  }, {
    name: '图标',
    enname: 'icon',
    id: 'icon',
    link: '/src/core/icon/demo/index'
  }, {
    name: '上下留白',
    enname: 'white-space',
    id: 'white-space',
    link: '/src/core/white-space/demo/index'
  }, {
    name: '左右留白',
    enname: 'wing-blank',
    id: 'wing-blank',
    link: '/src/core/wing-blank/demo/index'
  }, {
    name: '列表左滑',
    enname: 'swipe-action',
    id: 'swipe-action',
    link: '/src/core/swipe-action/demo/index'
  }, {
    name: '静态引导',
    enname: 'inpage-guide',
    id: 'inpage-guide',
    link: '/src/core/inpage-guide/demo/index'
  }, {
    name: '新功能推荐',
    enname: 'intro-board',
    id: 'intro-board',
    link: '/src/core/intro-board/demo/index'
  }, {
    name: '气泡引导',
    enname: 'popguide',
    id: 'popguide',
    link: '/src/core/popguide/demo/index'
  }, {
    name: '宫格',
    enname: 'grid',
    id: 'grid',
    link: '/src/core/grid/demo/index'
  }]
}, {
  title: '导航',
  theme: 'navigation',
  cardList: [{
    name: '选项卡',
    enname: 'tabs',
    id: ['tabs', 'tabs-content'],
    link: '/src/core/tabs/demo/index'
  }, {
    name: '分段器',
    enname: 'segmented-control',
    id: 'segmented-control',
    link: '/src/core/segmented-control/demo/index'
  }, {
    name: '面包屑',
    enname: 'breadcrumb',
    id: 'breadcrumb',
    link: '/src/core/breadcrumb/demo/index',
    "new": true
  }, {
    name: '导航栏',
    enname: 'nav-bar',
    id: 'nav-bar',
    link: '/src/core/nav-bar/demo/index',
    "new": true
  }, {
    name: '固钉',
    enname: 'affix',
    id: 'affix',
    link: '/src/core/affix/demo/index',
    "new": true
  }]
}, {
  title: '数据展示',
  type: 'data-display',
  cardList: [{
    name: '红点',
    enname: 'badge',
    id: 'badge',
    link: '/src/core/badge/demo/index'
  }, {
    name: '加载',
    enname: 'spin',
    id: 'spin',
    link: '/src/core/spin/demo/index'
  }, {
    name: '空页面',
    enname: 'empty',
    id: 'empty',
    link: '/src/core/empty/demo/index'
  }, {
    name: '骨架图',
    enname: 'skeleton',
    id: 'skeleton',
    link: '/src/core/skeleton/demo/index',
    "new": true
  }, {
    name: '列表',
    enname: 'list',
    id: ['list', 'list-item'],
    link: '/src/core/list/demo/index'
  }, {
    name: '通告栏',
    enname: 'notice-bar',
    id: 'notice-bar',
    link: '/src/core/notice-bar/demo/index',
    "new": true
  }, {
    name: '标签',
    enname: 'tag',
    id: 'tag',
    link: '/src/core/tag/demo/index',
    "new": true
  }, {
    name: '页码指示器',
    enname: 'page-indicator',
    id: 'page-indicator',
    link: '/src/core/page-indicator/demo/index',
    "new": true
  }, {
    name: '文本折叠',
    enname: 'textfold',
    id: 'textfold',
    link: '/src/core/textfold/demo/index',
    "new": true
  }, {
    name: '走马灯',
    enname: 'swiper',
    id: 'swiper',
    link: '/src/core/swiper/demo/index',
    "new": true
  }]
}, {
  title: '数据输入',
  type: 'data-entry',
  cardList: [{
    name: '搜索栏',
    enname: 'search-bar',
    id: 'search-bar',
    link: '/src/core/search-bar/demo/index'
  }, {
    name: '输入框',
    enname: 'input',
    id: 'input',
    link: '/src/core/input/demo/index'
  }, {
    name: '开关',
    enname: 'switch',
    id: 'switch',
    link: '/src/core/switch/demo/index'
  }, {
    name: '单选',
    enname: 'radio-group/radio-item',
    id: ['radio-group', 'radio-item'],
    link: '/src/core/radio-group/demo/index',
    "new": true
  }, {
    name: '多选',
    enname: 'checkbox-group(item)/checkbox',
    id: ['checkbox-group', 'checkbox-item', 'checkbox'],
    link: '/src/core/checkbox-group/demo/index',
    "new": true
  }, {
    name: '选择器',
    enname: 'picker',
    id: 'picker',
    link: '/src/core/picker/demo/index',
    "new": true
  }, {
    name: '级联选择器',
    enname: 'cascader-picker',
    id: 'cascader-picker',
    link: '/src/core/cascader-picker/demo/index',
    "new": true
  }, {
    name: '滑动条',
    enname: 'slider',
    id: 'slider',
    link: '/src/core/slider/demo/index',
    "new": true
  }]
}, {
  title: '反馈',
  type: 'feedback',
  cardList: [{
    name: '弹窗',
    enname: 'modal',
    id: 'modal',
    link: '/src/core/modal/demo/index'
  }, {
    name: '抽屉',
    enname: 'drawer',
    id: 'drawer',
    link: '/src/core/drawer/demo/index'
  }, {
    name: '可操作提示',
    enname: 'snackbar',
    id: 'snackbar',
    link: '/src/core/snackbar/demo/index'
  }]
}, {
  disabled: true,
  title: 'Coming Soon ～～ ↖(^ω^)↗',
  cardList: [{
    name: '图标',
    enname: 'icon',
    id: 'icon',
    link: '/src/core/icon/demo/index'
  }, {
    name: '上下留白',
    enname: 'white-space',
    id: 'white-space',
    link: '/src/core/white-space/demo/index',
    "new": true
  }, {
    name: '图标',
    enname: 'row',
    id: 'row',
    link: '/src/core/row/demo/index'
  }, {
    name: '图标',
    enname: 'col',
    id: 'col',
    link: '/src/core/col/demo/index'
  }]
}];
