// dingtalk-design-miniapp/es/picker/controlled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _fastDeepEqual = _interopRequireDefault(require("fast-deep-equal"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var _default = function _default() {
  var propsValue = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 'value';
  return {
    data: {
      cValue: null
    },
    didMount: function didMount() {
      var _this = this;
      this.cTrigger = function (v) {
        _this.setData({
          cValue: v
        });
      };
      this.cTrigger(this.props[propsValue]);
    },
    didUpdate: function didUpdate(prevProps) {
      var value = this.props[propsValue];
      if (!(0, _fastDeepEqual["default"])(prevProps[propsValue], value)) {
        this.cTrigger(value);
      }
    },
    methods: {
      cOnChange: function cOnChange(v) {
        var _this$props;
        if (!this.props.controlled) {
          this.cTrigger(v);
        }
        for (var _len = arguments.length, args = new Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
          args[_key - 1] = arguments[_key];
        }
        this.props.onChange && (_this$props = this.props).onChange.apply(_this$props, [v].concat(args));
      },
      cOnInput: function cOnInput(v) {
        var _this$props2;
        if (!this.props.controlled) {
          this.cTrigger(v);
        }
        for (var _len2 = arguments.length, args = new Array(_len2 > 1 ? _len2 - 1 : 0), _key2 = 1; _key2 < _len2; _key2++) {
          args[_key2 - 1] = arguments[_key2];
        }
        this.props.onInput && (_this$props2 = this.props).onInput.apply(_this$props2, [v].concat(args));
      }
    }
  };
};
exports["default"] = _default;
